arelle/plugin/validate/ROS/rules/ros.py

```diff
@@ -376,5 +376,6 @@ def rule_ros19(
             "ROS.19",
             message,
             modelObject=modelObjects,
+            principal_currency_values=principal_currency_values,
             principal_currency_value=principal_currency_value
```

Comment:
I think the `principal_currency_value` line still needs to be removed (this was presumably the cause of the error since thatvar is only defined if there's exactly one).

Reply:
`principal_currency_value` will show up as a None if it is not set.

Reply:
It'll throw a 'principal_currency_value' is not defined exception if the conditional check on line 359 is True because it's only assigned in the else branch.

Reply:
Can you update the PR description with the error you're trying to fix?  I assumed it was `NameError: name 'principal_currency_value' is not defined` (and that `KeyError` in the summary was a mistake), but perhaps I misunderstand the problem.

Reply:
I haven't seen the original exception, but I believe it does indeed throw a KeyError because of how this gets passed through and handled by the logger (it looks for an arg matching the name of the variable within the message).

Reply:
The `principal_currency_value` isn't defined in the "then" branch, so shouldn't the code fail (before and after the PR) with `NameError` on the `principal_currency_value=principal_currency_value` line before `Validation.warning` fails to resolve `%(principal_currency_values)s`?  It seems like we're missing a test.

Reply:
I restructured the validation to yield at the time of validation rather than the end of the validation.